Mathlib/Order/Hom/Set.lean

```diff
@@ -157,3 +157,97 @@ theorem compl_antitone : Antitone (compl : α → α) :=
 #align compl_antitone compl_antitone
 
 end BooleanAlgebra
+
+section Set
+
+variable {s t : Set α}
+
+/-- Equivalent types have order-isomorphic types of subsets. --/
+def Equiv.setOrderIso (e : α ≃ β) : Set α ≃o Set β where
+  toEquiv := Equiv.Set.congr e
+  map_rel_iff' := by simp
+
+@[simp] theorem Equiv.setOrderIso_toEquiv (e : α ≃ β) :
+    e.setOrderIso.toEquiv = Equiv.Set.congr e := rfl
+
+@[simp] theorem Equiv.setOrderIso_apply (e : α ≃ β) (s : Set α) :
+    e.setOrderIso s = e '' s := rfl
+
+@[simp] theorem Equiv.setOrderIso_apply_symm (e : α ≃ β) (s : Set β) :
+    e.setOrderIso.symm s = e.symm '' s := rfl
+
+@[simp] theorem Equiv.setOrderIso_subset_iff (e : α ≃ β) :
+    e.setOrderIso s ⊆ e.setOrderIso t ↔ s ⊆ t :=
+  e.setOrderIso.map_rel_iff
+
+@[simp] theorem Equiv.setOrderIso_symm_subset_iff (e : α ≃ β) {s t : Set β} :
+    e.setOrderIso.symm s ⊆ e.setOrderIso.symm t ↔ s ⊆ t :=
+  e.setOrderIso.symm.map_rel_iff
+
+/-- An injection from `α` to `β` gives one from `Set α` to `Set β`.  -/
+def Function.Embedding.setEmbedding (f : α ↪ β) : Set α ↪ Set β where
+  toFun := Set.image f
+  inj' := Set.image_injective.2 f.injective
+
+@[simp] theorem Function.Embedding.setEmbedding_apply (f : α ↪ β) (s : Set α) :
+    f.setEmbedding s = f '' s := rfl
+
+/-- An injection from `α` to `β` gives an order embedding from `Set α` to `Set β`.  -/
+def Function.Embedding.setOrderEmbedding (f : α ↪ β) : Set α ↪o Set β where
+  toEmbedding := f.setEmbedding
+  map_rel_iff' := fun {_} {_} ↦ Set.image_subset_image_iff f.injective
```

Comment:
I think you should just merge the former into the latter; we have quite a few other places in mathlib where we only have the order embedding.